From 679c0a5141c384a9a240f0217094899575309c5f Mon Sep 17 00:00:00 2001
From: Ding Yunliang <yunliang.ding@intel.com>
Date: Mon, 15 Feb 2016 15:29:34 +0800
Subject: [PATCH 1/3] atomisp2: fix NULL dereferenced issue

Change-Id: I0b9ed00d4fbf5391502a450e46a65eb00a37a905
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-20625
Original-Tracked-On: https://jira01.devtools.intel.com/browse/OAM-18598
Signed-off-by: Ding Yunliang <yunliang.ding@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/101598
Reviewed-by: Zaikuo Wang <zaikuo.wang@intel.com>
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Vns Murthy Sristi <vns.murthy.sristi@intel.com>
Build: Dessert M BXT RVP (SC)
Reviewed-by: HaiqiangX Yao <haiqiangx.yao@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: Ilkka Koskinen <ilkka.koskinen@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Signed-off-by: Yang Fan F <fan.f.yang@intel.com>
Reviewed-on: https://android.intel.com:443/471916
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
index a8cdb3e..71a97d9 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
@@ -1264,6 +1264,7 @@ static int atomisp_pci_probe(struct pci_dev *dev,
 	const struct atomisp_platform_data *pdata;
 	struct atomisp_device *isp;
 	unsigned int start;
+	void __iomem * const *table;
 	void __iomem *base;
 	int err, val;
 	u32 irq;
@@ -1299,11 +1300,12 @@ static int atomisp_pci_probe(struct pci_dev *dev,
 		return err;
 	}
 
-	base = pcim_iomap_table(dev)[ATOM_ISP_PCI_BAR];
-	if (!base) {
+	table = pcim_iomap_table(dev);
+	if (!table) {
 		dev_err(&dev->dev, "atomisp: error iomap table ptr\n");
 		return -EINVAL;
 	}
+	base = table[ATOM_ISP_PCI_BAR];
 
 	dev_dbg(&dev->dev, "base: %p\n", base);
 
-- 
1.9.1

